subprojects/ide-native/src/integTest/groovy/org/gradle/ide/xcode/XcodeSwiftExternalSourceDependenciesIntegrationTest.groovy

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.ide.xcode
+
+import org.gradle.ide.xcode.fixtures.AbstractXcodeIntegrationSpec
+import org.gradle.integtests.fixtures.build.BuildTestFile
+import org.gradle.internal.hash.HashUtil
+import org.gradle.nativeplatform.fixtures.app.SwiftAppWithLibrary
+import org.gradle.nativeplatform.fixtures.app.SwiftLib
+import org.gradle.nativeplatform.fixtures.app.SwiftLibTest
+import org.gradle.test.fixtures.file.TestFile
+import org.gradle.util.GFileUtils
+import org.gradle.vcs.fixtures.GitRepository
+import org.junit.Rule
+
+import static org.gradle.ide.xcode.internal.XcodeUtils.toSpaceSeparatedList
+
+class XcodeSwiftExternalSourceDependenciesIntegrationTest extends AbstractXcodeIntegrationSpec {
+    @Rule
+    GitRepository repo = new GitRepository('greeter', temporaryFolder.getTestDirectory())
+    BuildTestFile depProject
+
+    def "adds source dependencies Swift module of main component to Xcode indexer search path"() {
+        def fixture = new SwiftAppWithLibrary()
+
+        given:
+        buildTestFixture.withBuildInSubDir()
+        depProject = singleProjectBuild("greeter") {
+            buildFile << """
+                apply plugin: 'swift-library'
+            """
+            fixture.library.writeToProject(it)
+        }
+        def commit = repo.commit('initial commit', GFileUtils.listFiles(file('greeter'), null, true))
+
+        and:
+        settingsFile << """
+            sourceControl {
+                vcsMappings {
+                    withModule("org.test:greeter") {
+                        from vcs(GitVersionControlSpec) {
+                            url = "${repo.url}"
+                        }
+                    }
+                }
+            }
+        """
+        buildFile << """
+            apply plugin: 'swift-application'
+            apply plugin: 'xcode'
+            group = 'org.gradle'
+            version = '2.0'
+
+            dependencies {
+                implementation "org.test:greeter:latest.integration"
+            }
+        """
+        fixture.executable.writeToProject(testDirectory)
+
+        when:
+        succeeds 'xcode'
+
+        then:
+        executedAndNotSkipped(":xcodeProject", ":xcodeProjectWorkspaceSettings", ":xcodeSchemeAppExecutable", ":xcode")
```

Comment:
should build the binaries of `greeter` library

Reply:
you should update the test to show that the binaries of the `greeter` library are built. I'm not seeing any changes here.